internal/contour/endpointstranslator.go

```diff
@@ -244,6 +236,9 @@ func NewEndpointsTranslator(log logrus.FieldLogger) EndpointsInterface {
 // A EndpointsTranslator translates Kubernetes Endpoints objects into Envoy
 // ClusterLoadAssignment resources.
 type EndpointsTranslator struct {
+	// Observer notifies when the endpoints cache has been updated.
+	Observer Observer
```

Comment:
Why do we need a new observer interface? `Cond` already provides a way to get notifications on a change. You could add an object that does the channel wait on that and calls a function.

Reply:
Right but I'd need to add this to the workgroup and have this be long running. I think the Observer interface is much cleaner to implement & consume.